compute_tools/src/config.rs

```diff
@@ -92,6 +92,25 @@ pub fn write_postgres_conf(
         }
     }
 
+    // Check /proc/sys/vm/overcommit_memory -- if it equals 2 (i.e. linux memory overcommit is
+    // disabled), then the control plane has enabled swap and we should set
+    // dynamic_shared_memory_type = 'mmap'.
+    //
+    // This is (maybe?) temporary - for more, see https://github.com/neondatabase/cloud/issues/12047.
+    let overcommit_memory_contents = std::fs::read_to_string("/proc/sys/vm/overcommit_memory")
+        // ignore any errors - they may be expected to occur under certain situations (e.g. when
+        // not running in Linux).
```

Comment:
Seems like a good candidate for an `if cfg!(linux)`.

Reply:
Ah yup, good call. Done